test/unittests/commands/xmlupload/models/test_ingest.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+
+import pytest
+
+from dsp_tools.commands.xmlupload.models.ingest import DspIngestClient
+from dsp_tools.commands.xmlupload.models.ingest import IngestClient
+from dsp_tools.models.exceptions import BadCredentialsError
+from dsp_tools.models.exceptions import PermanentConnectionError
+from dsp_tools.models.exceptions import UserError
+
+
+@pytest.fixture()
+def dsp_ingest_url() -> str:
+    return "https://example.com"
+
+
+@pytest.fixture()
+def ingest_client(dsp_ingest_url: str) -> IngestClient:
+    return DspIngestClient(dsp_ingest_url, "token")
+
+
+@pytest.fixture()
+def shortcode() -> str:
+    return "0001"
+
+
+@pytest.fixture()
+def tmp_file(tmp_path: Path) -> Path:
+    return tmp_path / "filename.xml"
+
+
+def test_ingest_success(dsp_ingest_url, ingest_client, requests_mock, shortcode, tmp_file):  # type: ignore[no-untyped-def]
```

Comment:
wherever possible, I prefer proper typing:

```suggestion
def test_ingest_success(dsp_ingest_url: IngestClient, ingest_client: IngestClient, requests_mock: Mocker, shortcode: str, tmp_file: Path) -> None:
```

Reply:
IMHO for test code this is too noisy.

I like my types for production code, but for the test fixtures it is not really much of a help to me as these functions are only ever called by the test runner. WDYT?

Reply:
That's a valid point. But then you could accidentally pass objects of the wrong type into a function under test. Most of our production code doesn't verify the types of their input, because we assume that mypy prevents us from passing in wrong types. Or what's your opinion on that?

Reply:
`_ test_ingest_success` is the test function, i.e. the function that is doing the testing, I am not changing the type hints of the function under test which in this case is the function on the IngestClient.

Reply:
IMHO it is save for a function to assume the passed in arguments are of the proper type.
I do not think that every function should then do their type checks, that would too much code to maintain, mypy is doing this for us using the type hints.

Reply:
My point was a bit different: If the arguments of the test function are untyped, big parts of the body of the test function will also be untyped. And then the test function could accidentally pass wrong objects into the well-typed prod code, and noone will notice. This will deteriorate the quality of the tests.

IMHO, typing only protects us if we do it everywhere (?) As soon as parts of the (test) code are untyped, nasty things can find its way through the well-typed codebase incognito. And then we think that we tested something, but actually we didn't test it, because the tests are flawed.

Reply:
P.S. While I very much enjoy the current conversation, I want to stress that this is non-blocking for your PR ;-)